### ISSUE TITLE
CORS issue

### ISSUE THREAD

**Issue by itaditya** [prasmussen/glot-run#22]
Hi I'm unable to make an AJAX request to glot.io. I keep getting this error.

Failed to load https://run.glot.io/languages/python/latest: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8080' is therefore not allowed access. The response had HTTP status code 405.

**Comment by alokshakya** [comment]
Hi I am not able to call https://run.glot.io/languages/python/latest in Angular 5 application it gives following error.

Failed to load https://run.glot.io/languages/python/latest: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:4200' is therefore not allowed access. The response had HTTP status code 405.
![screenshot 226](https://user-images.githubusercontent.com/23319830/38577187-da95dc78-3d1d-11e8-8961-88b1a3d1d305.png)

**Comment by prasmussen** [comment]
Yeah, CORS is not supported at this time. I'm currently working on a new version of the api where I will consider allowing it. It will take at least a couple of months before the new api is ready though.

**Comment by sbstnssndn** [comment]
So we can't run code in glot from localhost? Is there a workaround for that?

**Comment by itaditya** [comment]
Create a backend server. From your frontend send a request to the server and from the server make a request to glot. For a working implementation check out https://github.com/itaditya/Code-Runner

**Comment by sbstnssndn** [comment]
Aaah I see, thanks. Sorry, why does this work? I thought if a request didn't work from the front, it wouldn't work from the back either.

**Comment by itaditya** [comment]
A request sent from browser i.e. Ajax is different from what is sent by the server. Since the request on the client side can be tampered by bad actors, it is less secure that is why cors exist to prevent sending requests from client side. Many Payment gateways also accept request from server only

**Comment by TheRealCatherine** [comment]
I want to vote strongly on allowing it because right now I'm pulling my hair out desperately looking for some shady 3rd party to do proxy POST requests to glot.io - and I fear that if I do even find such a shady 3rd party that works (most seem to only support GET) that when they are flaky or slow the "share to glot.io" button in our application isn't going to inspire the type of interest in glot.io that I was going for.

**Comment by sanjayme97** [comment]
is this fixed @prasmussen

**Comment by chumaumenze** [comment]
@sanjayme97 This might help you. https://github.com/Rob--W/cors-anywhere

**Comment by ten0s** [comment]
@prasmussen, I see some unused code for CORS
https://github.com/prasmussen/glot-run/blob/master/apps/glot/src/http/http_util.erl#L72-L83

Would you accept a PR?

**Comment by prasmussen** [comment]
> @prasmussen, I see some unused code for CORS
> https://github.com/prasmussen/glot-run/blob/master/apps/glot/src/http/http_util.erl#L72-L83
> 
> Would you accept a PR?

Sorry, but I won't do any changes to the current version in regards to CORS
